pymc/distributions/timeseries.py

```diff
@@ -237,64 +297,39 @@ def logp(self, x):
+        # Implement using AePPL
+        # I need to create a graph that calculates the logp of GRW
+        # I can use AePPL or PyMC to do it
 
-        Parameters
-        ----------
-        point: dict, optional
-            Dict of variable values on which random values are to be
-            conditioned (uses default point if not specified).
-        size: int, optional
-            Desired size of random sample (returns one sample if not
-            specified).
+        def normal_logp(value, mu, sigma):
+            logp = (
+                -0.5 * at.pow((value - mu) / sigma, 2)
+                - at.log(at.sqrt(2.0 * np.pi))
+                - at.log(sigma)
+            )
+            return logp
 
-        Returns
-        -------
-        array
-        """
-        # sigma, mu = distribution.draw_values([self.sigma, self.mu], point=point, size=size)
-        # return distribution.generate_samples(
-        #     self._random,
-        #     sigma=sigma,
-        #     mu=mu,
-        #     size=size,
-        #     dist_shape=self.shape,
-        #     not_broadcast_kwargs={"sample_shape": to_tuple(size)},
-        # )
-        pass
-
-    def _random(self, sigma, mu, size, sample_shape):
-        """Implement a Gaussian random walk as a cumulative sum of normals.
-        axis = len(size) - 1 denotes the axis along which cumulative sum would be calculated.
-        This might need to be corrected in future when issue #4010 is fixed.
+        # Create logp calculation graph the inital time point
+        innit_logp = normal_logp(value[0] - init, mu, sigma)
+
+        # Create logp calculation graph for innovations
+        stationary_vals = at.diff(value[1:]) - init
+        innov_logp = normal_logp(stationary_vals, mu, sigma)
+
+        """ A bunch of stuff that can be ignored
+        innit_logp = pm.logp(pm.Normal.dist(mu, sigma), value[:1] - init)
+        # https: // aesara.readthedocs.io / en / latest / library / tensor / extra_ops.html?highlight = at.diff
+        innov_logp = pm.logp(pm.Normal.dist(mu, sigma), at.diff(value))
+        # https: // numpy.org/doc/stable/ reference / generated / numpy.concatenate.html
         """
-        if size[len(sample_shape)] == sample_shape:
-            axis = len(sample_shape)
-        else:
-            axis = len(size) - 1
-        rv = stats.norm(mu, sigma)
-        data = rv.rvs(size).cumsum(axis=axis)
-        data = np.array(data)
-
-        # the following lines center the random walk to start at the origin.
-        if len(data.shape) > 1:
-            for i in range(data.shape[0]):
-                data[i] = data[i] - data[i][0]
-        else:
-            data = data - data[0]
-        return data
 
-    def _distr_parameters_for_repr(self):
-        return ["mu", "sigma"]
+        # Return both calculation logps in a vector. This is fine because somewhere
+        # down the line these will be summed together
+        total_logp = at.concatenate([innit_logp, innov_logp])
```

Comment:
This is not only fine but needed, for example for model comparison where you need the elemwise likelihood of the observations